### PR TITLE
feat: strengthen auth foundation defaults and validation

### PR DESCRIPTION
**Pull request by larimonious**
## Summary
- bump ntnt to 0.4.9
- rewrite DD-043 into a unified phased implementation plan and mark Phase 0 complete
- implement the first auth foundation improvements for Phase 1

## What changed
- `enable_auth(...)` now accepts the real 1-arg and 2-arg forms
- added stricter auth config validation with better type errors and unknown-key suggestions
- defaulted auth cookie security more safely based on env/site URL heuristics
- initialize configured auth session stores from the interpreter-side `enable_auth` path too
- added startup auth summary logging for enabled providers, routes, session TTL, and cookie posture
- added regression tests for config parsing and sqlite session-store edge cases
- regenerated docs and bumped version metadata to 0.4.9

## Verification
- `cargo test -q`
- `cargo build --release --locked`
- `./target/release/ntnt docs --generate`
- two review passes, including Codex review, with follow-up fixes for arity registration and session-store initialization

## Notes
This PR is the foundation slice, not the route-protection/session-helper/staged-auth phases yet. It is meant to make the auth entry point safer, clearer, and easier to debug before the heavier cleanup phases land.

### CODE REVIEW

File: src/interpreter.rs [hunk 1]
```diff
@@ -7182,61 +7196,224 @@ impl Interpreter {
...
+            _ => Err(IntentError::type_error(format!(
+                "enable_auth() providers must be a provider or provider array, got {}",
+                value.type_name()
+            ))),
+        }
+    }
+
+    fn default_auth_cookie_secure(&self) -> bool {
+        if let Ok(env) = std::env::var("NTNT_ENV") {
+            if env.eq_ignore_ascii_case("development") {
```
[the remark this answers is not in the document]
Comment (larimonious) [reply src/interpreter.rs]: Fixed. The cookie-secure heuristic is now shared and recognizes both `NTNT_ENV=dev` and `NTNT_ENV=development`, plus localhost-style `SITE_URL` values, so the interpreter and stdlib paths stay consistent.

File: src/interpreter.rs [hunk 2]
```diff
@@ -7182,61 +7196,224 @@ impl Interpreter {
...
+                        };
+                        crate::stdlib::auth::SessionStore::Sqlite(path.to_string())
+                    } else if s.starts_with("postgres:") || s.starts_with("postgresql:") {
+                        crate::stdlib::auth::SessionStore::Postgres(s.clone())
+                    } else if s.starts_with("redis:") || s.starts_with("valkey:") {
+                        crate::stdlib::auth::SessionStore::Redis(s.clone())
+                    } else {
+                        return Err(IntentError::type_error(format!(
+                            "enable_auth() config[\"session_store\"] must be one of: memory, sqlite:PATH, postgres://..., redis://..., valkey://..."
+                        )));
```
[the remark this answers is not in the document]
Comment (larimonious) [reply src/interpreter.rs]: Fixed. `session_store` parsing is now centralized through strict helpers, so malformed prefixes no longer silently pass the interpreter path. This also normalizes empty `sqlite:` to `./sessions.db` instead of accepting an empty path.

File: src/stdlib/auth.rs [hunk 3]
```diff
@@ -6351,35 +6364,12 @@ pub fn init() -> HashMap<String, Value> {
...
-                        }
-                    }
-                    SessionStore::Memory => {}
+                if let Err(e) = initialize_session_store(&session_store) {
+                    eprintln!("[auth] Failed to initialize session store: {}", e);
+                    return Err(IntentError::runtime_error(format!(
+                        "Failed to initialize session store: {}",
+                        e
+                    )));
                 }
```
[the remark this answers is not in the document]
Comment (larimonious) [reply src/stdlib/auth.rs:6485]: Fixed. I aligned the stdlib `enable_auth` path with the new validation/defaults and also made auth routes auto-register at server start when an auth config is present, so imported/module usage no longer silently misses route setup while the interpreter path does more work.

File: src/interpreter.rs [hunk 4]
```diff
@@ -7182,61 +7196,224 @@ impl Interpreter {
...
+                config.providers = self.parse_auth_providers(map.get("providers").unwrap())?;
+                for (key, value) in map {
+                    if key == "providers" {
+                        continue;
+                    }
+                    self.apply_auth_option(&mut config, key, value)?;
                 }
-                if let Some(Value::Int(i)) = map.get("session_ttl") {
-                    config.session_ttl = *i;
+            }
```
[the remark this answers is not in the document]
Comment (larimonious) [reply src/interpreter.rs:7214]: Fixed. A single config map without `providers` now errors explicitly with `enable_auth() config map must include a "providers" array` instead of falling through to the generic provider parsing error.

File: src/stdlib/auth.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (larimonious) [reply src/stdlib/auth.rs:6485]: Follow-up fix pushed in `6cf3531`: the interpreter-side `enable_auth` config path now also accepts `refresh_ttl`, with a regression test, so the stdlib and interpreter paths stay aligned on that option too.

File: src/interpreter.rs [hunk 5]
```diff
@@ -8919,6 +9096,12 @@ impl Interpreter {
         use std::sync::Arc;
         use std::thread;
 
+        if let Some(auth_config) = crate::stdlib::auth::get_auth_config() {
+            if !self.server_state.has_route("GET", "/auth/{provider}") {
```
[the remark this answers is not in the document]
Comment (larimonious) [reply src/interpreter.rs]: Fixed in `48a3c31` with the same route-setup/idempotency change applied to the async server path too. I reran tests, release build, and docs generation after the fix.

File: src/stdlib/auth.rs [hunk 6]
```diff
@@ -6195,7 +6251,7 @@ pub fn init() -> HashMap<String, Value> {
     //
     // Session storage options: "memory" (default), "sqlite:./path.db", "postgres://url", or "redis://url".
     // @param providers Array of provider configs created by oauth() or oauth_discover()
-    // @param options Optional map with keys: session_secret, session_ttl, after_login, after_logout, session_store
+    // @param options Optional map with keys: session_secret, session_ttl, success_url/after_login, failure_url/after_failure, logout_url/after_logout, cookie_name, cookie_secure, session_store, store_tokens
```
[the remark this answers is not in the document]
Comment (larimonious) [reply src/stdlib/auth.rs]: Fixed in `48a3c31`. The stdlib `enable_auth` option docs now include `refresh_ttl`, and docs were regenerated afterward.

File: src/stdlib/auth.rs [hunk 7]
```diff
@@ -6263,151 +6320,144 @@ pub fn init() -> HashMap<String, Value> {
...
-                    .and_then(|v| match v {
-                        Value::String(s) => Some(s.clone()),
-                        _ => None,
-                    })
-                    .unwrap_or_else(|| DEFAULT_SESSION_SECRET_SENTINEL.to_string());
+                let get_option = |keys: &[&str]| {
+                    options
+                        .as_ref()
+                        .and_then(|opts| keys.iter().find_map(|key| opts.get(*key)))
+                };
```
[the remark this answers is not in the document]
Comment (larimonious) [reply src/stdlib/auth.rs:6387]: Fixed in `48a3c31`. The stdlib `enable_auth` path now validates option keys against an explicit allowlist and returns a typed error with a suggestion instead of silently ignoring unknown keys.

File: src/interpreter.rs [hunk 8]
```diff
@@ -7182,61 +7196,196 @@ impl Interpreter {
...
+            other => {
+                let suggestion = self
+                    .auth_option_suggestion(other)
+                    .map(|s| format!(" Did you mean \"{}\"?", s))
+                    .unwrap_or_default();
+                return Err(IntentError::type_error(format!(
+                    "enable_auth() unknown config key \"{}\".{}",
+                    other, suggestion
+                )));
             }
```
[the remark this answers is not in the document]
Comment (larimonious) [reply src/interpreter.rs:7374]: Fixed in `48a3c31`. Beyond the earlier `refresh_ttl` follow-up, I also tightened the stdlib/interpreter alignment further: auth route setup is now idempotent and the stdlib path now rejects unknown options instead of silently accepting drift.

File: src/interpreter.rs [hunk 9]
```diff
@@ -8492,6 +8663,12 @@ impl Interpreter {
             None => (env_port.unwrap_or(port), false, None),
         };
 
+        if let Some(auth_config) = crate::stdlib::auth::get_auth_config() {
+            if !self.server_state.has_route("GET", "/auth/{provider}") {
+                self.setup_auth_routes(&auth_config)?;
```
[the remark this answers is not in the document]
Comment (larimonious) [reply src/interpreter.rs]: Fixed in `48a3c31`. Server start now always calls `setup_auth_routes()` when auth config exists, and that setup is idempotent, so imported/module auth usage no longer depends on a single route already existing.
